src/ITILValidationTemplate.php

```diff
@@ -0,0 +1,174 @@
+                // are not correctly processed. Indeed, document item corresponding to the destination item is not created and
+                // image src is not containing the ITIL item foreign key, so image will not be visible for helpdesk profiles.
+                // As fixing it is really complex (requires lot of refactoring in images handling, both on JS and PHP side),
+                // it is preferable to disable usage of images in templates.
+                'disable_images' => true,
+            ]
+        ];
+    }
+
+    /**
+     * Display specific "approver" field
+     *
+     * @param $value
+     * @param $name
+     * @param $options
+     */
+    public static function displayValidatorField(
+        $value = null,
+        $options = []
+    ) {
+        if ($value) {
+            $itemtype = array_key_first($value);
+            $items_id = array_map(function ($item) {
+                return $item['items_id'];
+            }, $value[$itemtype]);
+        }
+
+        $options['users_id_requester'] = \Session::getLoginUserID();
+        $options['itemtype_target']    = $itemtype ?? null;
+        $options['groups_id']          = $value ? $value[$itemtype][0]['groups_id'] : null;
+        $options['items_id_target']    = $value && $itemtype == 'Group' && count($items_id) == 1 ?
+            $items_id[0] : $items_id ?? null;
+        $options['right']              = 'validate_request';
+        $options['display']            = false;
+
+        return CommonITILValidation::dropdownValidator($options);
+    }
+
+    public function displaySpecificTypeField($id, $field = [], array $options = [])
+    {
+        if ($field['name'] == 'approver') {
+            echo self::displayValidatorField(ITILValidationTemplate_Target::getTargets($id), []);
+        }
+    }
+
+    public static function getSpecificValueToSelect($field, $name = '', $values = '', array $options = [])
+    {
+        if ($field == 'approver') {
+            return self::displayValidatorField($values[$field], $options);
+        }
+
+        return parent::getSpecificValueToSelect($field, $name, $values, $options);
+    }
+
+    public function rawSearchOptions()
+    {
+        $tab = parent::rawSearchOptions();
+
+        $tab[] = [
+            'id'                 => '4',
+            'name'               => __('Content'),
+            'field'              => 'content',
+            'table'              => $this->getTable(),
+            'datatype'           => 'text',
+            'htmltext'           => true
+        ];
+
+        return $tab;
+    }
+
+    public static function getIcon()
+    {
+        return "fas fa-layer-group";
+    }
+
+    public function postTargets()
+    {
+        $itilValidationTemplatesTarget = new ITILValidationTemplate_Target();
+        $itilValidationTemplatesTarget->deleteByCriteria([
+            'itilvalidationtemplates_id' => $this->getID(),
+        ]);
```

Comment:
This should be moved in the `if` block, otherwise, a PATCH from API that would juste redefine the `name` property would remove all defined targets.